src/Transports.AspNetCore.SystemTextJson/GraphQLRequestDeserializer.cs

```diff
@@ -24,34 +24,40 @@ public class GraphQLRequestDeserializer : IGraphQLRequestDeserializer
 
         public GraphQLRequestDeserializer(Action<JsonSerializerOptions> configure)
         {
+            // Add converter that deserializes Variables property
             _serializerOptions.Converters.Add(new ObjectDictionaryConverter());
 
             configure?.Invoke(_serializerOptions);
         }
 
         public async Task<GraphQLRequestDeserializationResult> DeserializeFromJsonBodyAsync(HttpRequest httpRequest, CancellationToken cancellationToken = default)
         {
+            var bodyReader = httpRequest.BodyReader;
+
             JsonTokenType jsonTokenType;
             try
             {
-                jsonTokenType = await PeekJsonTokenTypeAsync(httpRequest.BodyReader, cancellationToken).ConfigureAwait(false);
+                jsonTokenType = await PeekJsonTokenTypeAsync(bodyReader, cancellationToken);
             }
             catch (JsonException)
             {
                 // Invalid request content, assign None so it falls through to WasSuccessful = false
                 jsonTokenType = JsonTokenType.None;
             }
 
+            if (cancellationToken.IsCancellationRequested)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+            }
```

Comment:
```c#
 /// <summary>Throws a cancellation exception if cancellation has been requested via <paramref name="cancellationToken"/>.</summary>
        /// <param name="cancellationToken">The token to check for a cancellation request.</param>
        internal static void ThrowIfCancellationRequested(CancellationToken cancellationToken)
        {
            if (cancellationToken.IsCancellationRequested)
            {
                ThrowOperationCanceledException(innerException:null, cancellationToken);
            }
        }
```

Reply:
Where does `ThrowOperationCanceledException` method come from?

Reply:
Guessing you might've copied that from somewhere else. Take a look at what I did and see if it's all g.

Reply:
I just demonstrate you MS code. It is implementation of `ThrowIfCancellationRequested` so no need to call `if (cancellationToken.IsCancellationRequested)` on your side.

Reply:
🤦‍♂ I must've been tired.